.github/actions/setup/action.yml

```diff
@@ -0,0 +1,36 @@
+name: Setup Rust Environment
+
+inputs:
+  key:
+    description: Cache key
+    required: true
+  toolchain:
+    description: Pass-through to toolchain on actions-rs
+    default: stable
+    required: false
+  components:
+    description: Pass-through to components on actions-rs
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - uses: actions-rs/toolchain@v1
+      id: toolchain-install
+      with:
+        profile: minimal
+        override: true
+        toolchain: ${{ inputs.toolchain }}
+        components: ${{ inputs.components }}  
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: ${{ inputs.key }}-${{ runner.os }}-${{ inputs.toolchain }}-${{ steps.toolchain-install.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock', '**/Cargo.toml') }}
+        restore-keys: |
+          ${{ inputs.key }}-${{ runner.os }}-${{ inputs.toolchain }}-${{ steps.toolchain-install.outputs.rustc_hash }}-
+          ${{ inputs.key }}-${{ runner.os }}-${{ inputs.toolchain }}-
```

Comment:
@Kijewski  - believe the `toolchain` is used as part of the rustc_hash output, are you sure you want both?

https://github.com/actions-rs/toolchain#outputs

Reply:
So that when a the patch version of rustc is released, not everything has to be built anew.